### PR TITLE
Determine patients with signed careplans within 122 days of enrollment

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: app/models/health/agency_performance.rb [hunk 1]
```diff
@@ -16,7 +16,10 @@ def initialize range:, agency_scope: nil
 
     def agency_counts
       @agency_counts ||= agencies.map do |id, name|
-        patient_ids = patient_referrals.select{ |_, agency_id| agency_id == id }.keys
+        patient_ids = patient_referrals.select do |_, referral|
```
Comment (eanders) [inline app/models/health/agency_performance.rb]: This might be cleaner as `do |_, (agency_id, _)|`

File: app/models/health/agency_performance.rb [hunk 2]
```diff
@@ -30,6 +32,12 @@ def agency_counts
         with_signed_careplans = careplan_dates.select{ |p_id, _| p_id.in?(patient_ids) }.keys
         without_signed_careplans = patient_ids - with_signed_careplans
 
+        with_careplans_in_122_days = patient_ids.select do |p_id|
+          careplan_date = careplan_dates[p_id]&.to_date
+          enrollment_date = patient_referrals[p_id][1]
+          careplan_date.present? && (careplan_date - enrollment_date).to_i <= 122
```
Comment (eanders) [inline app/models/health/agency_performance.rb]: @awisema, per our recent call with BHCHP, `careplan_date` should be the date of the QA that represents the signature.

File: app/models/health/agency_performance.rb [hunk 3]
```diff
@@ -247,5 +247,15 @@ def qualifying_activity_dates
...
     end
 
+    def qa_signature_dates
+      @qa_signatures ||= Health::QualifyingActivity.submittable.
+        distinct.
+        where(patient_id: patient_referrals.keys). # limit to patients in scope
+        where(activity: :pctp_signed).
+        where(date_of_activity: @range).
+        order(date_of_activity: :asc).
+        pluck(:patient_id, :date_of_activity).to_h
```
Comment (eanders) [inline app/models/health/agency_performance.rb]: Should this be `group(:patient_id).maximum(:date_of_activity)`?

Comment (awisema) [reply app/models/health/agency_performance.rb]: Yep.

File: app/models/health/agency_performance.rb [hunk 4]
```diff
@@ -234,5 +247,12 @@ def qualifying_activity_dates
         pluck(:patient_id, :date_of_activity).to_h
     end
 
+    def qa_signature_dates
+      @qa_signatures ||= Health::QualifyingActivity.submittable.
+        where(patient_id: patient_referrals.keys). # limit to patients in scope
+        where(activity: :pctp_signed).
+        group(:patient_id).maximum(:date_of_activity)
```
Comment (eanders) [inline app/models/health/agency_performance.rb]: You lost the range.  Is that not necessary any more?

Comment (awisema) [reply app/models/health/agency_performance.rb]: Oops -- yeah, I think that should still be there...

**Pull request merged**